src/utils.js

```diff
@@ -150,6 +152,21 @@ function altContigName(contig: string): string {
   }
 }
 
+/**
+ * Pipe all promise events through a deferred object.
+ * This is similar to deferred.resolve(promise), except that it allows progress
```

Comment:
I believe this should be exactly the same as `deferred.resolve(promise)`, which resolves/rejects the deferred with whatever the promise eventually winds up as. Except for the notifications part.